bindings/pyroot/pythonizations/python/ROOT/_pythonization/_uhi.py

```diff
@@ -1,23 +1,296 @@
 # Author: Silia Taider CERN  03/2025
 
 ################################################################################
-# Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.                      #
+# Copyright (C) 1995-2025, Rene Brun and Fons Rademakers.                      #
 # All rights reserved.                                                         #
 #                                                                              #
 # For the licensing terms see $ROOTSYS/LICENSE.                                #
 # For the list of contributors see $ROOTSYS/README/CREDITS.                    #
 ################################################################################
 from __future__ import annotations
-from abc import ABC, abstractmethod
+
 import enum
-from typing import Tuple, Iterator, Union, Callable, Any
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Iterator, Tuple, Union
+
+"""
+Implementation of the module level helper functions for the UHI
+"""
+
+
+def _underflow(hist: Any, axis: int) -> int:
+    return 0
+
+
+def _overflow(hist: Any, axis: int) -> int:
+    return _get_axis(hist, axis).GetNbins() + 1
+
+
+class _loc:
+    def __init__(self, *values: float) -> None:
+        self.values = values
+        self.offset = 0
+
+    def __add__(self, other: int) -> _loc:
+        if isinstance(other, int):
+            self.offset += other
+            return self
+        raise TypeError("Unsupported type for addition")
+
+    def __sub__(self, other: int) -> _loc:
+        if isinstance(other, int):
+            self.offset -= other
+            return self
+        raise TypeError("Unsupported type for subtraction")
+
+    def __call__(self, hist: Any, axis: int) -> int:
+        return _get_axis(hist, axis).FindBin(*self.values) + self.offset
+
+
+class _rebin:
+    def __init__(self, ngroup):
+        self.ngroup = ngroup
+
+    def __call__(self, hist):
+        rebin_methods = {2: "Rebin2D", 3: "Rebin3D"}
+        rebin_method = rebin_methods.get(hist.GetDimension(), "Rebin")
+        rebin_method = getattr(hist, rebin_method)
+        return rebin_method(*self.ngroup, newname=hist.GetName())
+
+
+class _sum:
+    def __call__(self, hist, axis):
+        dim = hist.GetDimension()
+
+        if isinstance(axis, int):
+            axis = (axis,)
+        if dim == 1:
+            return hist.Integral()
```

Comment:
This makes sense to me and I think it's also what the UHI specifies, but I have a feeling we need to document this bit particularly well (together with the rest of the documentation for the whole UHI implementation)